File: leetcode/js/shuffle-string.js
/**
 * @param {string} s
 * @param {number[]} indices
 * @return {string}
 */
var restoreString = function (s, indices) {
  let result = Array.from(s);

  for (let i = 0; i < indices.length; i++) {
    const idx = indices[i];
    result[idx] = s[i];
  }

  return result.join("");
};

File: leetcode/rust/shuffle-string.rs
impl Solution {
    pub fn restore_string(s: String, indices: Vec<i32>) -> String {
        let mut result = vec![' '; s.len()];

        for (&i, c) in indices.iter().zip(s.chars()) {
            result[i as usize] = c;
        }

        result.iter().collect()
    }
}
